src/workshops/4/pgr.c

```diff
@@ -0,0 +1,53 @@
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+enum { TWO = 2, FIVE = 5, NINE = 9 };
+
+/* Denne funktion skal returnere 1 hvis x er et primtal og 0 ellers */
+int isPrime(int number) {
+  int result = 1;
+  for (int i = 2; i < number / 2; i++) {
+    if (number % i == 0) {
+      result = 0;
+      break;
+    }
+  }
+  return result;
+}
+
+/* Denne funktion skal returnere 1 hvis gcd(x,2)=1 og 0 ellers */
+int isGcd1(int number) { return number; }
+
+/* Denne funktion skal returnere 1 hvis 9^x-2 mod 5 = 2 og 0 ellers */
+int is2mod5(int number) {
+  int result = ((int)pow(NINE, number % FIVE) - TWO) % FIVE == TWO ? 1 : 0;
```

Comment:
@MadsHeilmann Very difficult to read and understand. Also, if you want to have a real one-liner, no need to declare a variable first and then return it in line 25. You can just write:
```suggestion
return ((int)pow(NINE, number % FIVE) - TWO) % FIVE == TWO ? 1 : 0;
```
But please make it a bit more readable first.